src/client/services/cdc/index.js

```diff
@@ -2,26 +2,14 @@ const io = require('socket.io-client');
 let socket = io.connect('/');
 
 const CDC = {
-  initGraphSocket(updateFunction) {
-    socket.on('layoutPackage', (cyZip) => {
-      let msg = atob(cyZip);
-      try {
-        let cyJSON = JSON.parse(msg);
-        updateFunction(cyJSON);
-      } catch(err) {
-        if (msg.length > 0) {
-          console.log(JSON.parse(msg));
-          //console.log(msg);
-        } else {
-          throw err;
-        }
+  getGraphAndLayout(uri, version) {
+    return fetch(`http://${window.location.hostname}/get-graph-and-layout?uri=${encodeURIComponent(uri)}&version=${version}`, {
+      method: 'GET', headers: {
+        'Content-type': 'application/json',
+        'Accept': 'application/json'
       }
-    });
-  },
-
-  // Request a graph from the server. Should send back a cyJSON graph
-  requestGraph(uri, version) {
-    socket.emit('getLayout', {uri: uri, version: version.toString()});
+    }).then(res => res.json())
+    .then(data => JSON.parse(data));
```

Comment:
Redundant code.  res.json() already parses the response body.

https://developer.mozilla.org/en-US/docs/Web/API/Body/json

Reply:
Resolved. Turns out we had left in a stringify on the server-side, which is why we still needed that on the frontend.